src/components/link/index.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { ElementType, FC, ReactNode } from 'react';
+import { chakra, useMultiStyleConfig } from '@chakra-ui/react';
+
+interface Props {
+  children: ReactNode;
+  as?: ElementType;
+  leftIcon?: ElementType;
+  rightIcon?: ElementType;
+  isDisabled?: boolean;
+  isExternal?: boolean;
+  [key: string]: unknown;
+}
+
+const Link: FC<Props> = ({
+  as = chakra.a,
+  children,
+  isDisabled = false,
+  isExternal = false,
+  leftIcon,
+  rightIcon,
+  ...rest
+}) => {
+  const styles = useMultiStyleConfig(`Link`);
+
+  const Component = chakra(as, {
```

Comment:
Use [Chakra Factory](https://chakra-ui.com/docs/styled-system/features/chakra-factory) to convert non-chakra components to chakra-enabled components (For example, `Link` from `react-router-dom`